src/Taboo/TabooRoom.py

```diff
@@ -111,48 +111,48 @@ def __processJoin(self, qmsg):
 
 
     def joinPlayer(self, ws, playerName, teamNumber):
-        player, team = self.getPlayer(playerName)
+        player = self.getPlayer(playerName)
 
         # A new player joins
         if not player:
-            if teamNumber not in self.teams:
-                #A new team
-                self.teams[teamNumber] = TabooTeam(teamNumber)
-            team = self.teams.get(teamNumber)
+            team = self.__getTeam(teamNumber)
             player = TabooPlayer(self.txQueue, playerName, team)
-            self.__finalizeJoin(ws, player, team.teamNumber)
+            self.__finalizeJoin(ws, player)
             #self.processEvent(PlayerJoin(player))
             return True
 
-        assert team is not None, "What happened!"
         #new join-request for an existing player will be accepted
         #but we ignore the requested team and associate the player to the original team
         if teamNumber != team.teamNumber:
             trace(Level.warn, "Player {} forced to join their "
                 "original team {}".format(player.name, team.teamNumber))
 
-        self.__finalizeJoin(ws, player, team.teamNumber)
+        self.__finalizeJoin(ws, player)
         return True
 
     def getPlayer(self, playerName):
-        """ Returns (player, team)
+        """ Returns player
 
             player: player matching the name
-            team: team it belongs to
-
-            (None, None) if no player exists by that name
+                None if no player exists by that name
         """
         for t in self.teams.values():
             v = t.getPlayer(playerName)
             if v is not None:
-                return (v, t)
-        return (None, None)
+                return v
+        return None
 
-    def __finalizeJoin(self, ws, player, teamNumber):
+    def __finalizeJoin(self, ws, player):
         player.addConn(ws)
         self.playerByWs[ws] = player
-        self.txQueue.put_nowait(ClientTxMsg(["JOIN-OKAY", player.name, teamNumber],
+        self.txQueue.put_nowait(ClientTxMsg(["JOIN-OKAY", player.name, player.team.teamNumber],
             {ws}, initiatorWs=ws))
 
     def spectatorCount(self):
         return sum(1 for plyr in self.playerByWs.values() if not plyr)
+
+    def __getTeam(self, teamNumber):
+        team = self.teams.get(teamNumber)
+        if not team:
+            return min(self.teams.values(), lambda t: len(t.members))
```

Comment:
Are you sure this is okay? I tried this:

```
>>> class Num:
...    def __init__(self, x):
...       self.x = x
... 
>>> a = Num(1)
>>> b = Num(2)
>>> c = Num(1)
>>> min([a, b, c], lambda t: t.x)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: '<' not supported between instances of 'function' and 'list'
>>> 
```

This seems to think that "lambda t: ..." is an element to compare as well!

Reply:
Here is the fix:

the second argument should be `key=lambta t: len(t.members)`  (we call out that this is the key function by name). It works then!

Reply:
I was trying out what editing in the browser looks like because it offers to do that. Tried that and it allowed me to add "key=" here so just made the change. Resolving this comment :)